docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+In my case, I only like the message of the first commit, so I will delete the others.
+
+Press `i` to place the editor into `INSERT` mode, and prepare the single commit message.
+
+![submit_pr_squash_commits_8.png](./imgs/submit_pr_squash_commits_8.png)
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+
+The interactive `rebase` command finally completes.
+
+![submit_pr_squash_commits_9.png](./imgs/submit_pr_squash_commits_9.png)
+
+
+Now it is time to push the changes from the local feature branch to the remote feature branch.
+
+Type :
+
+```
+git push -f
+```
+
+You can confirm that there is only 1 commit by typing:
+
+```
+git log
+```
+
+![submit_pr_squash_commits_11.png](./imgs/submit_pr_squash_commits_11.png)
+
+or via your browser in your remote feature branch.
+
+![submit_pr_squash_commits_10.png](./imgs/submit_pr_squash_commits_10.png)
+
+
+### 17. Creating the pull request (PR)
+
+Now that you have completed the `rebase` and `commit squashing` steps, it is finally time to create the pull request.
+
+In your browser, navigate to your remote `feature` branch, and press the `Compare & pull request` button.
+
+![submit_pr_create_pr_1.png](./imgs/submit_pr_create_pr_1.png)
+
+The short and long description fields of the pull request are auto-filled using the short and long description of your commit.
+
+![submit_pr_create_pr_2.png](./imgs/submit_pr_create_pr_2.png)
+
+Fill in as much extra information as possible to properly describe the purpose of the pull request ( for example, fixing a specific bug or implementing an extension point, etc).
+
+(The title of the pull request should start with “WIP” for “work in progress” when the pull request is not complete yet.)
+
+ Press the `Create pull request` button.
+
+
+
+![submit_pr_create_pr_3.png](./imgs/submit_pr_create_pr_3.png)
+
+The pull request is created, and some continuous integration jobs commence.
+
+![submit_pr_create_pr_4.png](./imgs/submit_pr_create_pr_4.png)
+
+Make certain all these jobs complete successfully.
+
+### 18. Agreeing to the contributor license agreement (CLA)
+
+You must agree to the contributor license agreement (CLA) before the pull request can be `merged`. Please read
+[Agreeing to the CLA](https://loopback.io/doc/en/contrib/code-contrib.html#agreeing-to-the-cla).
+
+### 19. Go through the PR review process
+
+Request a code review or document review.
+
+Add a specific committer to speed up this process. (for example, @bajtos, @raymondfeng).
+
+The review process is iterative, so it is normal that multiple extra commits may be required until the pull request is finally accepted.
+
+
+### 20. Final rebase and squashing of commits
+
+Once the pull request is finally approved, repeat the `Rebase` and the `Squashing Commits` sections so that your pull request has a single, clean final commit and corresponding message.
```

Comment:
Only should repeat if there was something they changed.

Reply:
@nabdelgadir . It is possible that nothing changed in my number of commit messages, but that my forked repo is behind on many commits from the original repo, and so a rebase is needed.

But I agree that my paragraph should specify that these steps need only be performed IF it is necessary.

So I will change the paragraph to : 
"
Once the pull request is finally approved, repeat the `Rebase` and the
`Squashing Commits` sections, if necessary, so that your pull request has a single, clean, final commit and corresponding message.
"